dbms/src/Functions/LeastGreatest.h

```diff
@@ -120,5 +121,363 @@ class FunctionVectorizedLeastGreatest : public IFunction
 private:
     const Context & context;
 };
+template <bool least, bool use_collator = true>
+struct LeastGreatestStringImpl
+{
+    static void processImpl(
+        const TiDB::TiDBCollatorPtr & collator,
+        size_t a_size,
+        size_t b_size,
+        const unsigned char * a_data,
+        const unsigned char * b_data,
+        ColumnString::Chars_t & c_data,
+        ColumnString::Offsets & c_offsets,
+        int i)
+    {
+        int res = 0;
+        StringRef s_ra(&a_data[0], a_size);
+        StringRef s_rb(&b_data[0], b_size);
+        auto pre_offset = StringUtil::offsetAt(c_offsets, i);
+        if constexpr (use_collator)
+            res = collator->compare(reinterpret_cast<const char *>(&a_data[0]), a_size, reinterpret_cast<const char *>(&b_data[0]), b_size);
+
+        else
+            res = memcmp(&a_data[0], &b_data[0], std::min(a_size, b_size));
+
+        if constexpr (least)
+        {
+            if (res < 0)
+            {
+                memcpy(&c_data[pre_offset], &a_data[0], a_size);
```

Comment:
string's terminating zero is not added?

Reply:
Yes, we don't need it.

Reply:
> Yes, we don't need it.

The original column strings are terminated with zero, why we don't need it here?